qiita_db/sql_connection.py

```diff
@@ -1012,6 +1014,20 @@ def execute_fetchindex(self, idx=-1):
         """
         return self.execute()[idx]
 
+    def _cmds_executor(self, cmds, func_str):
+        error_msg = []
+        for cmd in cmds:
+            try:
+                cmd()
+            except Exception as e:
+                error_msg.append(str(e))
+        self._post_commit_funcs = []
```

Comment:
Once the commit/rollback is done, the transaction object should remain clean. Hence, once the functions have been executed, we are resetting these attributed. These two lists are mutually exclusive, only one of them is executed, so when one of them is executed, we can reset both of them.